packages/backend/src/managers/applicationManager.spec.ts

```diff
@@ -1358,3 +1358,59 @@ describe('pod detection', async () => {
     expect(mocks.removePodMock).toHaveBeenCalledWith('engine-1', 'pod-1');
   });
 });
+
+describe('getImageTag', () => {
+  const manager = new ApplicationManager(
+      '/path/to/user/dir',
+      {} as GitManager,
+      taskRegistry,
+      {
+        postMessage: mocks.postMessageMock,
+      } as unknown as Webview,
+      {
+        onPodStart: mocks.onPodStartMock,
+        onPodStop: mocks.onPodStopMock,
+        onPodRemove: mocks.onPodRemoveMock,
+        startupSubscribe: mocks.startupSubscribeMock,
+        onMachineStop: mocks.onMachineStopMock,
+      } as unknown as PodmanConnection,
+      {
+        getRecipeById: vi.fn().mockReturnValue({ name: 'MyRecipe' } as Recipe),
+      } as unknown as CatalogManager,
+      {} as ModelsManager,
+      {} as TelemetryLogger,
+      localRepositoryRegistry,
+    );
+  test('return recipe-container tag if container image prop is not defined', () => {
+    const recipe = {
+      id: 'recipe1',
+    } as Recipe;
+    const container = {
+      name: 'name'
+    } as ContainerConfig;
+    const imageTag = manager.getImageTag(recipe, container);
+    expect(imageTag).equals('recipe1-name:latest');
+  });
+  test('return container image prop is defined', () => {
+    const recipe = {
+      id: 'recipe1',
+    } as Recipe;
+    const container = {
+      name: 'name',
+      image: 'quay.io/repo/image:v1',
+    } as ContainerConfig;
+    const imageTag = manager.getImageTag(recipe, container);
+    expect(imageTag).equals('quay.io/repo/image:v1');
+  });
+  test('append latest tag to container image prop if it has no tag', () => {
+    const recipe = {
+      id: 'recipe1',
+    } as Recipe;
+    const container = {
+      name: 'name',
+      image: 'quay.io/repo/image',
+    } as ContainerConfig;
+    const imageTag = manager.getImageTag(recipe, container);
+    expect(imageTag).equals('quay.io/repo/image:latest');
+  });
+})
```

Comment:
looks the file is unformatted, don't know if we need pre-commit hooks like in podman-desktop

Reply:
yes i guess we should add it

Reply:
opened issue https://github.com/projectatomic/ai-studio/issues/603